integ/operator-test-app/docker-entrypoint.sh

```diff
@@ -13,21 +13,13 @@ drop_privs_cmd() {
     fi
 }
 
-envsubst < /usr/local/flink-conf.yaml > $FLINK_HOME/conf/flink-conf.yaml
-
-# As the taskmanager pods are accessible only by (cluster) ip address,
-# we must manually configure this based on the podIp kubernetes
-# variable, which is assigned to TASKMANAGER_HOSTNAME env var by the
-# operator.
-if [ -n "$TASKMANAGER_HOSTNAME" ]; then
-    echo "taskmanager.host: $TASKMANAGER_HOSTNAME" >> "$FLINK_HOME/conf/flink-conf.yaml"
-fi
-
 # Add in extra configs set by the operator
 if [ -n "$OPERATOR_FLINK_CONFIG" ]; then
```

Comment:
Another option is to do

```
+if [ "$FLINK_DEPLOYMENT_TYPE" = "jobmanager" ]; then
+    echo "jobmanager.rpc.address: $HOST_NAME" >> "$FLINK_HOME/conf/flink-conf.yaml"
+fi
+
 # As the taskmanager pods are accessible only by (cluster) ip address,
 # we must manually configure this based on the podIp kubernetes
+if [ "$FLINK_DEPLOYMENT_TYPE" = "taskmanager" ]; then
+    echo "taskmanager.host: $HOST_IP" >> "$FLINK_HOME/conf/flink-conf.yaml"
 fi
```

And remove that logic from the operator.

Reply:
My preference is to have it in the operator (as you've done), as this gives us more flexibility (the operator is much easier to change than the images).